pkg/providers/vsphere/vmlifecycle/bootstrap_cloudinit.go

```diff
@@ -49,9 +49,10 @@ func BootStrapCloudInit(
 		sshPublicKeys = strings.Join(cloudInitSpec.SSHAuthorizedKeys, "\n")
 	}
 
-	uid := string(vmCtx.VM.UID)
-	// Use the existing instance ID if it is already set on the VM.
-	// This ensures cloud-init uses the same instance ID across B/R.
+	uid := cloudInitSpec.InstanceID
+	if uid == "" {
+		uid = string(vmCtx.VM.UID)
+	}
```

Comment:
I suppose this is for backward compatibility to avoid updating an existing VM's cloud-init instance ID, right? It would be great if you could add a comment for clarity and future reference.

Reply:
Right, added comment.